pkg/R/DataFrame.R

```diff
@@ -253,6 +253,96 @@ setMethod("unpersist",
             x
           })
 
+#' Repartition
+#'
+#' Return a new DataFrame that has exactly numPartitions partitions.
+#'
+#' @param x A SparkSQL DataFrame
+#' @param numPartitions The number of partitions to use.
+#' @rdname repartition
+#' @export
+#' @examples
+#'\dontrun{
+#' sc <- sparkR.init()
+#' sqlCtx <- sparkRSQL.init(sc)
+#' path <- "path/to/file.json"
+#' df <- jsonFile(sqlCtx, path)
+#' newDF <- repartition(df, 2L)
+#'}
+
+setGeneric("repartition", function(x, numPartitions) { standardGeneric("repartition") })
+
+#' @rdname repartition
+#' @export
+setMethod("repartition",
+          signature(x = "DataFrame", numPartitions = "integer"),
```

Comment:
We know the argument name in this specific case, but I can't imagine `numPartitions` is the only time we'll ever need to use this trick. Being able to write `5` instead of `5L` is convenient in any function where we need to send an integer, but there's no guarantee that those function arguments will always be named `numPartitions`. By using `sys.call`, we can use `numToInt` in any number of functions and know that the warning message will always be relevant to the actual function being called.

Good call on `as.integer`. I've changed that part.